compositor_pipeline/src/pipeline/whip_whep/whip_handlers/create_whip_session.rs

```diff
@@ -0,0 +1,217 @@
+            "Cannot lock input connections".to_string(),
+        ));
+    }
+
+    peer_connection.on_track(Box::new(move |track, _, _| {
+        let track_kind = track.kind();
+        let state_clone = state.clone();
+        let input_id_clone = input_id.clone();
+        let video_sender_clone = input_components.video_sender.clone();
+        let audio_sender_clone = input_components.audio_sender.clone();
+        let depayloader_clone = input_components.depayloader.clone();
+
+        tokio::spawn(async move {
+            match track_kind {
+                RTPCodecType::Video => {
+                    if let Some(sender) = video_sender_clone {
+                        handle_track(
+                            track,
+                            state_clone,
+                            input_id_clone,
+                            depayloader_clone,
+                            sender,
+                        )
+                        .await;
+                    }
+                }
+                RTPCodecType::Audio => {
+                    if let Some(sender) = audio_sender_clone {
+                        handle_track(
+                            track,
+                            state_clone,
+                            input_id_clone,
+                            depayloader_clone,
+                            sender,
+                        )
+                        .await;
+                    }
+                }
+                _ => {
+                    debug!("RTPCodecType not supported!")
+                }
+            }
+        });
+        Box::pin(async {})
+    }));
+
+    peer_connection.on_ice_connection_state_change(Box::new(move |state| {
+        info!("ICE connection state changed: {state:?}");
+        Box::pin(async {})
+    }));
+
+    let description = RTCSessionDescription::offer(offer)?;
+
+    peer_connection.set_remote_description(description).await?;
+    let answer = peer_connection.create_answer(None).await?;
+
+    peer_connection.set_local_description(answer).await?;
+
+    gather_ice_candidates_for_one_second(peer_connection.clone()).await;
+
+    let Some(sdp) = peer_connection.local_description().await else {
+        return Err(WhipServerError::InternalError(
+            "Read local description error".to_string(),
+        ));
+    };
+    debug!("Sending SDP answer: {sdp:?}");
+
+    let body = Body::from(sdp.sdp.to_string());
+    let response = Response::builder()
+        .status(StatusCode::CREATED)
+        .header("Content-Type", "application/sdp")
+        .header("Access-Control-Expose-Headers", "Location")
+        .header("Location", format!("/session/{}", encode(&id)))
+        .body(body)?;
+    Ok(response)
+}
+
+fn get_whip_input_connection_options(
+    state: Arc<WhipWhepState>,
+    input_id: InputId,
+) -> Result<WhipInputConnectionOptions, WhipServerError> {
+    if let Ok(connections) = state.input_connections.lock() {
```

Comment:
unwrap on mutex is fine, I don't think it is worth complexity to handle  it everywhere. Mutex lock only fails if it is poisoned (some different thread needs to panic for this to happen)